delfin/alert_manager/trap_receiver.py

```diff
@@ -155,10 +153,17 @@ def _mib_builder(self):
         try:
             self.mib_view_controller = view.MibViewController(mib_builder)
 
-            # set mib path to mib_builder object and load mibs
-            mib_path = builder.DirMibSource(self.snmp_mib_path),
-            mib_builder.setMibSources(*mib_path)
+            # Append custom mib path to default path for loading mibs
+            mib_sources = mib_builder.getMibSources() + (builder.DirMibSource(
+                self.snmp_mib_path),)
+            mib_builder.setMibSources(*mib_sources)
             if len(MIB_LOAD_LIST) > 0:
+                # Compile custom mib files (.mib) which will be used by pysnmp
+                # http://mibs.snmplabs.com/asn1/ for satisfying dependencies
+                compiler.addMibCompiler(mib_builder,
+                                        sources=[self.snmp_mib_path,
+                                                 'http://mibs.snmplabs.com'
```

Comment:
it will load from internet?

Reply:
Right, it is used for resolving dependencies. Current method uses precompiled mibs (.py) for using which I feel is not good as user has to do it externally

Reply:
In general, delfin is installed under internal environment which can not access the internet directly

Reply:
Ok , let me update to use precompiled files as earlier, just fix the path issue
Will keep and issue to continue to check for other alternatives

Reply:
Updated the PR